cmd/dockerd/daemon.go

```diff
@@ -424,6 +424,11 @@ func newRouterOptions(ctx context.Context, cfg *config.Config, d *daemon.Daemon,
 		return routerOptions{}, err
 	}
 
+	var compression string
+	if cfg.Features["zstd"] {
+		compression = "zstd"
+	}
+
```

Comment:
```suggestion
	var compression string
	if cfg.Features["zstd"] {
		compression = "zstd"
	}

```

IMHO, it would be nicer to have a daemon config `default-compression` that could be set to `zstd`.
Also, in a follow up we could add a `--compress` switch to `docker build` that would allow to override it on a per-build basis.

This would also be a partial workaround for longer build times (due to compression) with the containerd integration - user could opt-out of the compression to save build time.

Reply:
AFAIK, adding `--compress` to `docker push` was rejected, not `docker build`.

Reply:
I think I am missing something... `docker build` does not produce an image, isn't it? That's why there was a request to add `--compress` to `docker push`. Because image only starts to exist during push.

Reply:
That's different with containerd integration.

With graphdrivers, `docker build` produces uncompressed image and is stored in the Docker image store without the OCI image representation. Only when you do `docker push`, the OCI manifest is produced and the image layer is compressed.

With containerd, `docker build` produces both the compressed OCI image representation (OCI index/manifest and compressed blobs) and the uncompressed (unpacked) image. So `docker push` just pushes the image 1:1 (so the image ID is the same).

Reply:
Oh, understood.